src/context-parser-handlebars.js

```diff
@@ -894,6 +963,28 @@ ContextParserHandlebars.prototype.handleRawBlock = function(input, i, saveToBuff
     throw "Parsing error! Cannot encounter '}}}}' close brace of raw block.";
 };
 
+/**
+* @function ContextParserHandlebars.reset
+*
+* @description
+* All non-config internal variables are needed to reset!
+*/
+ContextParserHandlebars.prototype.reset = function() {
+    /* save the processed char */
+    this._buffer = [];
+
+    /* save the char/line no being processed */
+    this._charNo = 0;
+    this._lineNo = 1;
+    this._fileName = '';
+
+    /* expose the processed partial cache */
+    this.processedPartialCache = {};
```

Comment:
can you clarify why `this.processedPartialCache = {};` has to be reset? 
should it be shared across multiple templates?